application/pendingtx/pending_test.go

```diff
@@ -268,7 +267,7 @@ func mustDelTx(t *testing.T, hndlr *Handler, tx *objs.Tx) {
 }
 
 func setup(t *testing.T) (*Handler, *mockTrie, func()) {
-	dir, err := ioutil.TempDir("", "badger-test")
+	dir, err := os.MkdirTemp("", "badger-test")
```

Comment:
Prefer [testing#T.TempDir](https://pkg.go.dev/testing#T.TempDir) for tests. It will handle cleanup and and checking `err` for you already.

Also for all subsequent instances in this PR.